### DISCUSSION TITLE
Value binding for radio button not updating

### DISCUSSION THREAD

**Discussion by azophy** [alpinejs/alpine#2702] [7. Bugs (archived)] [answered]
I'm having a trouble when I tried to update `value` attribute of my dynamically generated list of radio buttons. I have tried to look up AlpineJS documentation regarding this problem but found no working solutions. Any help is appreciated. Thanks

## Minimal reproducable code

codepen: https://codepen.io/azophy/pen/ExbdZBN

```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
</head>
<body>
    <script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>
    <div x-data="{
      current_index: 0,
      selected_id: null,

      option_groups: [
        [
          { id: 'A', label: 'Alice' },
          { id: 'B', label: 'Bob' },
          { id: 'C', label: 'Charlie' },
        ],
        [
          { id: 'D', label: 'Diana' },
          { id: 'E', label: 'Eli' },
          { id: 'F', label: 'Fox' },
        ],
      ],
    }">
      <template x-for="(item, iteration) in option_groups[current_index]" :key="iteration">
        <p>
          <input type="radio" :value="item.id" x-model="selected_id">
          <span x-text="item.label"></span>
        </p>
      </template>

      <p>
        current_index: <input type="number" x-model="current_index" min=0 max=1>
        <span x-text="'selected_id ' + selected_id"></span>
      </p>
    </div>
</body>
</html>
```

## Expectation

The value attribute of the radio input would be updated when `current_index` value is changed. 

## What I found

When I changed `current_value`, the radio labels updated accordingly. However the `value` attribute did not. When I selected the radio button, the selected_id still showing A-C values instead of D-F.

**Comment by azophy** [comment, marked as answer]
Huge thanks to @SimoTod for helping me regarding this issue. It turns out that I've picked a wrong key. Using array index as `x-for` key should be avoided, instead try to pick a more unique value. In this case I only need to replace `:key="iteration"` with `:key="item.id"`.

Updated codepen: https://codepen.io/azophy/pen/podxaJw
Related documentation: https://alpinejs.dev/directives/for#keys

**Reply by cxyguopong** [reply]
great!

**Reply by Alionides** [reply]
Thanks this helped me as well ))
